assignment2.py

```diff
@@ -0,0 +1,155 @@
+class ListNode:
+    def __init__(self, value):
+        self.next = None
+        self.value = value
+        
+def string_compression(s):
+    count = 1
+
+    compressed = []
+
+    for i in range(1, len(s)):
+        if s[i - 1] == s[i] :
+            count += 1
+
+        else:
+            compressed.append(s[i - 1])
+            compressed.append(str(count))
+            #reset count
+            count = 1
+
+    compressed.append(s[-1])
+    #append conpressed amt
+    compressed.append(str(count))
+
+    compressed_str = ''.join(compressed)
+    #checkcase for if its og
+    return compressed_str if len(compressed_str) < len(s) else s
+
+def add_linked_lists_reverse(l1, l2):
+    carry = 0
+
+    dummy_head = ListNode(0)
+    current = dummy_head
+
+    while l1 or l2:
+        x = l1.value if l1 else 0
+        y = l2.value if l2 else 0
+
+        total = x + y + carry
+        carry = total // 10
+        current.next = ListNode(total % 10)
+        current = current.next
+        if l1:
+            l1 = l1.next
+        if l2:
+            l2 = l2.next
+
+    if carry > 0:
+        current.next = ListNode(carry)
+
+    return dummy_head.next
+
+def delete_middle_node(node):
+    if node is None or node.next is None:
```

Comment:
Be careful using `is` over `==` in Python. `is` is used for [identity equality](https://www.blog.pythonlibrary.org/2017/02/28/python-101-equality-vs-identity/#:~:text=Hopefully%20by%20now%20you%20understand,object%20you%20are%20referring%20to.). Also, because `None` is already falsey in Python, you don't need to explicitly check against it here.
```suggestion
    if not node or not node.next:
```